arvind.py

```diff
@@ -124,32 +131,49 @@ def download(self, download_dir="./"):
             # Note the format specification is important so we get mp4 and not taller than 480
             'format': "bestvideo[height<=480][ext=mp4]+bestaudio[ext=m4a]/best[height<=480][ext=mp4]"
         }
-        with youtube_dl.YoutubeDL(ydl_options) as ydl:
-            pp = pprint.PrettyPrinter()
-            try:
-                ydl.add_default_info_extractors()
-                # vinfo = ydl.extract_info(self.url, download=True)
-                vinfo = ydl.extract_info(self.url, download=True)
-                # Save the remaining "temporary scraped values" of attributes with actual values
-                # from the video metadata.
-                self.uid = vinfo.get('id', '')
-                self.title = vinfo.get('title', '')                             
-                self.description = vinfo.get('description', '')                             
-
-                # Set the filepath and thumbnail attributes of the video object.
-                self.filepath = self.set_filepath_and_thumbnail(vinfo, download_dir=download_dir)
-                
-                if not vinfo['license']:
-                    self.license = "Licensed not available"
-                elif vinfo['license'].find("Creative Commons") != -1:
-                    self.license_common = True
-                else:
-                    self.license = vinfo['license']
-
-            except (youtube_dl.utils.DownloadError,
-                    youtube_dl.utils.ContentTooShortError,
-                    youtube_dl.utils.ExtractorError,) as e:
-                    print('==> Error downloading videos', e)
-                    # pp.pprint(e)
-                    return False
+
+        match = YOUTUBE_ID_REGEX.match(self.url)
+        if not match:
+            print('==> URL ' + self.url + ' does not match YOUTUBE_ID_REGEX')
+            return False
+        youtube_id = match.group('youtube_id')
+
+        vinfo_json_path = os.path.join(YOUTUBE_CACHE_DIR, youtube_id+'.json')
+        # First try to get from cache:
+        if os.path.exists(vinfo_json_path):
+            vinfo = json.load(open(vinfo_json_path))
+
+        # else get using youtube_dl:
+        else:
+            with youtube_dl.YoutubeDL(ydl_options) as ydl:
+                pp = pprint.PrettyPrinter()
+                try:
+                    ydl.add_default_info_extractors()
+                    # vinfo = ydl.extract_info(self.url, download=True)
+                    vinfo = ydl.extract_info(self.url, download=download)
```

Comment:
Here, `download` is now set to `False` so the call `ydl.extract_info` will only get the video info json data and not download the actual file. Runs very fast...